tests/cnf/ran/oran/tests/oran-provision.go

```diff
@@ -1 +1,135 @@
 package tests
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/eco-goinfra/pkg/configmap"
+	"github.com/openshift-kni/eco-goinfra/pkg/ocm"
+	"github.com/openshift-kni/eco-goinfra/pkg/oran"
+	"github.com/openshift-kni/eco-goinfra/pkg/reportxml"
+	"github.com/openshift-kni/eco-goinfra/pkg/secret"
+	"github.com/openshift-kni/eco-goinfra/pkg/siteconfig"
+	. "github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/raninittools"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/oran/internal/helper"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/oran/internal/tsparams"
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ContinueOnError is deliberately left out of this Ordered container. If the invalid ProvisioningRequest does not
+// become valid again, we cannot test provisioning with a valid ProvisioningRequest.
+var _ = Describe("ORAN Provision Tests", Label(tsparams.LabelProvision), Ordered, func() {
+	var prBuilder *oran.ProvisioningRequestBuilder
+
+	AfterAll(func() {
+		By("verifying the spoke 1 kubeconfig path is set")
+		Expect(RANConfig.Spoke1Kubeconfig).
+			ToNot(BeEmpty(), "KUBECONFIG environment variable must be set to save spoke 1 kubeconfig")
+
+		By("saving the spoke 1 admin kubeconfig")
+		kubeconfigSecret, err := secret.Pull(HubAPIClient, RANConfig.Spoke1Name+"-admin-kubeconfig", RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get the spoke 1 kubeconfig secret")
+
+		kubeconfig, exists := kubeconfigSecret.Definition.Data["kubeconfig"]
+		Expect(exists).To(BeTrue(), "Kubeconfig key does not exist in kubeconfig secret")
+
+		err = os.WriteFile(RANConfig.Spoke1Kubeconfig, kubeconfig, 0644)
+		Expect(err).ToNot(HaveOccurred(), "Failed to save the spoke 1 admin kubeconfig")
+	})
+
+	// 77393 - Apply a ProvisioningRequest with missing required input parameter
+	It("recovers provisioning when invalid ProvisioningRequest is updated", reportxml.ID("77393"), func() {
+		By("creating a ProvisioningRequest with invalid policyTemplateParameters")
+		prBuilder = helper.NewProvisioningRequest(HubAPIClient, tsparams.TemplateValid).
+			WithTemplateParameter(tsparams.PolicyTemplateParamsKey, map[string]any{
+				// By using an integer when the schema specifies a string we can create an invalid
+				// ProvisioningRequest without being stopped by the webhook.
+				tsparams.TestName: 1,
+			})
+
+		var err error
+		prBuilder, err = prBuilder.Create()
```

Comment:
Since we rely on cleanup done by external code via CI, we probably should check if the env is clean in BeforeAll or BeforeEach for this test case, and skip if not?